kf/bpf.go

```diff
@@ -1104,10 +1104,19 @@ func (b *BPF) RemoveMapFiles(ifaceName string) error {
 		} else {
 			mapFilename = filepath.Join(b.hostConfig.BpfMapDefaultPath, ifaceName, k)
 		}
+		if !strings.HasPrefix(mapFilename, b.hostConfig.BpfMapDefaultPath) {
+			return fmt.Errorf("malicious mapFilename path")
+		}
 		if err := v.Unpin(); err != nil {
-			return fmt.Errorf("BPF program %s prog type %s ifacename %s map %s:failed to pin the map err - %#v",
+			return fmt.Errorf("BPF program %s prog type %s ifacename %s map %s:failed to unpin the map: %v",
 				b.Program.Name, b.Program.ProgType, ifaceName, mapFilename, err)
 		}
+		if fileExists(mapFilename) {
```

Comment:
If a BPF map is shared by multiple programs, and one program unpins the map while another program is still accessing it, it can lead to unpredictable behavior. To avoid this, you can check if the reference count of the map is less than 1 using `v.GetRefCount()`, and then proceed to delete the file.

Reply:
Is v.GetRefCount() is library function ?